probe-rs/src/probe/mod.rs

```diff
@@ -219,7 +218,9 @@ impl Probe {
 
     /// Leave debug mode
     pub fn detach(&mut self) -> Result<(), DebugProbeError> {
-        self.inner.detach()
+        self.inner.detach()?;
+        self.attached = false;
```

Comment:
Probably these two actions should be reversed

Reply:
awhy do you think so? I if the detach fails, we don't want to go into detached mode, right?

Reply:
I think that detach is not an atomic operation and it can fail halfway. Currently, `self.attached==true` means "the probe is probably attached", and if you reverse two lines, it will be "the probe is definitely attached".

Reply:
Hmm well, I see your point! But I am not sure that's the way to go. The detach function is just for structural cleanup and no physical, protocol wise actions are required afaik.

I am unsure now :D

Reply:
Well, I don't have strong feelings about this anyway, just wanted to express my thoughts :)

Reply:
Yes, I appreciate that a lot :) I am unsure what the best option is atm. We also need some way to lock the probe settings somehow. Maybe a builder would be good for this.